### PR TITLE
Feature – Upgrade final-form

### LINKED ISSUE #473: Upgrade final-form
This causes a lot of compilation errors

### PR DESCRIPTION
**Pull request by webmaster128**
`FieldValidator` is now a generic, so we need to adapt some of our code.

Closes #473

### CODE REVIEW

File: packages/bierzo-wallet/src/routes/payment/components/ReceiverAddress/index.tsx [hunk 1]
```diff
@@ -18,8 +18,8 @@ interface Props {
 }
 
 const ReceiverAddress = ({ form, selectedChainCodec }: Props): JSX.Element => {
-  const recipientValidator: FieldValidator = React.useMemo(() => {
-    const validator = (value: string): string | undefined => {
+  const recipientValidator: FieldValidator<string | undefined> = React.useMemo(() => {
```
Comment (albertandrejev) [inline packages/bierzo-wallet/src/routes/payment/components/ReceiverAddress/index.tsx]: You can use `InputValue` here as well, if I understand you correctly.

File: packages/sanes-chrome-extension/src/routes/restore-account/components/SetMnemonicForm.tsx [hunk 2]
```diff
@@ -14,11 +14,9 @@ import { RESTORE_ACCOUNT } from "../../paths";
 
 export const MNEMONIC_FIELD = "mnemonicField";
 
-const englishMnemonicValidator: FieldValidator = (value): string | undefined => {
-  if (typeof value !== "string") throw new Error("Input must be a string");
-
+const englishMnemonicValidator: FieldValidator<string | undefined> = (value): string | undefined => {
```
Comment (albertandrejev) [inline packages/sanes-chrome-extension/src/routes/restore-account/components/SetMnemonicForm.tsx]: `InputValue` can be used here too.

File: packages/bierzo-wallet/src/routes/payment/components/ReceiverAddress/index.tsx (hunk 1 above)
Comment (webmaster128) [reply packages/bierzo-wallet/src/routes/payment/components/ReceiverAddress/index.tsx]: Right, thanks, will update

File: packages/sanes-chrome-extension/src/routes/restore-account/components/SetMnemonicForm.tsx (hunk 2 above)
Comment (webmaster128) [reply packages/sanes-chrome-extension/src/routes/restore-account/components/SetMnemonicForm.tsx]: Right, thanks, will update

**Pull request merged**